### DISCUSSION TITLE
NUGHUDs

### DISCUSSION THREAD

**Discussion by MrAlaux** [MrAlaux/Nugget-Doom#54] [Show and tell]
I guess we can share our NUGHUDs here. 
**Try to keep all your submissions in a single comment**; edit it if it needs to be updated, and feel free to reply to it to announce updates.
**Comments announcing updates or otherwise not providing any content will be eventually hidden**, to prioritize the posts that do provide content.

I'll start off with my own:

### [**Cramped NUGHUD**](https://github.com/MrAlaux/Nugget-Doom/files/14676098/NUGHUD-Cramped.zip) (last updated 20/Mar/2024)

<details>
<summary>Preview</summary>

![nugg0003](https://github.com/MrAlaux/Nugget-Doom/assets/73968015/52fb53d0-2faf-481d-959f-90a586411cea)
</details>

### [**DSDA NUGHUD**](https://github.com/MrAlaux/Nugget-Doom/files/14174147/NUGHUD-DSDA.zip)

<details>
<summary>Preview</summary>

![nugg0002](https://github.com/MrAlaux/Nugget-Doom/assets/73968015/a121435d-2e2b-4b0d-ad92-f64de826cbbc)
</details>

**Comment by liPillON** [comment]
Here are mine.

## Current GitHub projects
These are actively developed and supported. 
All NUGHUD features up until Nugget-Doom 2.3.0 have been been implemented.
Work is underway to add support for recent features, as of the 3.1.0 release.

### [**nughud_zdoom**](https://github.com/liPillON/nughud_zdoom/releases/latest) (2024-05-21)

<details>
<summary>preview: </summary>

![nughud_zdoom_readme](https://raw.githubusercontent.com/liPillON/nughud_zdoom/main/README.png)

</details>

### [**nughud_split**](https://github.com/liPillON/nughud_split/releases/latest) (2024-05-19)

<details>
<summary>preview: </summary>

![nughud_split_readme](https://raw.githubusercontent.com/liPillON/nughud_split/main/NUGHUD_split_classic.png)

</details>


### [**nughud_fullstbar**](https://github.com/liPillON/nughud_fullstbar/releases/latest) (2024-05-21)

<details>
<summary>preview: </summary>

![nughud_fullstbar_readme](https://raw.githubusercontent.com/liPillON/nughud_fullstbar/main/README.png)

</details>



## Previous GitHub projects
These do not leverage most recent NUGHUD features.
They still works as intended up to nugget-doom 2.3.0 release.

### [**nughud_modern**](https://github.com/liPillON/nughud_modern/releases/latest)

<details>
<summary>preview: </summary>

![nughud_modern_readme](https://raw.githubusercontent.com/liPillON/nughud_modern/main/README.png)

</details>



### [**nughud_minimal**](https://github.com/liPillON/nughud_minimal/releases/latest)

<details>
<summary>preview: </summary>

![nughud_minimal_readme](https://raw.githubusercontent.com/liPillON/nughud_minimal/main/README.png)

</details>

<details>
<summary>supports gzdoom, too: </summary>

![nughud_minimal_sbarinfo](https://raw.githubusercontent.com/liPillON/nughud_minimal/main/SBARINFO.png)

</details>



## Proof of Concepts
These were discussed while the NUGHUD features were dropping and were shared [here](https://github.com/MrAlaux/Nugget-Doom/issues/26) and [here](https://github.com/MrAlaux/Nugget-Doom/issues/18)
They haven't been tested in a while, I don't expect them to support every NUGHUD feature properly.
I don't know if/when I'll work on them again.

### [**nughud-interdoom**](https://github.com/MrAlaux/Nugget-Doom/files/9846104/nughud-interdoom.zip) (aka "Crispy+")

<details>
<summary>preview: </summary>

![nugud-interdoom-screenblocks12](https://user-images.githubusercontent.com/48625273/197383821-cd375ae9-2b15-4a2d-959c-0761042900e4.png)

</details>

### [**nughud-stacked**](https://github.com/MrAlaux/Nugget-Doom/files/9851847/nughud-poc.zip) (aka "pillar")

<details>
<summary>preview: </summary>

![nugud-stacked-screenblocks12](https://user-images.githubusercontent.com/48625273/197533031-2f13ca1a-e0c1-423b-a888-9d8dcafea65a.png)

</details>



## Experiments
These were posted on Doomworld as a preview but I then lost interest in them for various reason.
I don't know if/when I'll work on them again.

### [nughud_experiment.zip](https://github.com/MrAlaux/Nugget-Doom/files/12266633/nughud_experiment.zip)

<details>
<summary>preview: </summary> 

![nughud_experiment](https://github.com/MrAlaux/Nugget-Doom/assets/48625273/dddcbfe5-e5ed-40f6-933d-aa03bafe4895) 

</details>

**Comment by ak-988** [comment]
[Classic-ish Status bar](https://github.com/MrAlaux/Nugget-Doom/files/12174343/NUGHUD-classic.zip)

It's just the same information as the vanilla status bar, including the face, which is surprisingly hard to find among ports. 

<details>
<summary>preview</summary>

![screenshot_nugget-testing_oldbean](https://github.com/MrAlaux/Nugget-Doom/assets/76888018/f63da129-6791-4482-82ac-a00d10d9bed6)

</details>

[4:3 version](https://github.com/user-attachments/files/15964742/4.3.nughud.zip)

Here's a 4:3 version based on Sarnion's awesome HUD.

<details>
<summary>preview</summary>

![screenshot_nugget-doom_h8slaught](https://github.com/MrAlaux/Nugget-Doom/assets/76888018/f23c7573-e705-4afc-9815-6b4520059815)

</details>

I also use this [brightmaps lump](https://github.com/MrAlaux/Nugget-Doom/files/12660433/brghtmps.zip) to remove brightmaps from items. I think it's cheesy to see hidden cell pickups and stuff like that.

**Comment by liPillON** [comment]


**Reply by MrAlaux** [reply]
Any chance you could update your first post and include this HUD there? As stated in the OP (which you might want to re-read as I just updated it), I'd rather there's just one post per user to minimize clutter. No pressure, though - I'm not completely certain regarding how I want to manage this either, I'm not familiar with GH Discussions, so I'm definitely at fault. I'll hide these posts later.

**Comment by d-bind** [comment]
### [nughud_qlike](https://github.com/MrAlaux/Nugget-Doom/files/12503881/nughud_qlike.zip)

A minimalist Quake-style HUD with dynamic health/armor icons. Based on @liPillON's nughud_experiment.

<details><summary>preview</summary>

![doom2](https://github.com/MrAlaux/Nugget-Doom/assets/104466974/77ce91cf-6738-4d10-be14-836b9eff9293)
![tnt](https://github.com/MrAlaux/Nugget-Doom/assets/104466974/b49f8617-1ec9-44f2-a415-c1b023f25f5b)
</details>

The custom icons [won't work well](https://github.com/MrAlaux/Nugget-Doom/assets/104466974/5cb0dbea-c292-4b24-8515-2470390569a9) with wads that shuffle the palette around (AA, BtSX, DotB, etc), though tinting is fine. A compatibility version without dynamic icons is also included.

<details><summary>compat preview</summary>

![compat](https://github.com/MrAlaux/Nugget-Doom/assets/104466974/e84443f4-05e2-4eac-ae79-6d66f25430f8)
</details>

**Comment by BeemerJ** [comment]
# [wideHUD Legacy](https://github.com/MrAlaux/Nugget-Doom/files/13987521/wideHUD.zip)

A lil' vanilla styled (minus a mugshot) HUD for widescreen resolutions of all scales 👍
<details>
<summary>Preview:</summary>
<br>
<IMG src="https://github.com/MrAlaux/Nugget-Doom/assets/74435632/3f9246e7-1f11-4c82-b5d0-c84c576672b2">
</details>

_Tested with Nugget Doom 2.3.0..._

-----
# [wideHUD_v2](https://github.com/user-attachments/files/16045755/wideHUD_v2.zip)

**Version 2**

Includes universal custom STBAR support. The _downside_ is that I had to remove the "ARMS / FRAGS" section and replaced it with a mugshot (So I'll be leaving the previous version up for those who prefer only having a vanilla hud).
<details>
<summary>Preview:</summary>
<br>
<IMG title="Valiant" src="https://github.com/MrAlaux/Nugget-Doom/assets/74435632/c25e69f5-b4fa-4b60-b675-e58caca95236">
<IMG title="Zeppelin Armada" src="https://github.com/MrAlaux/Nugget-Doom/assets/74435632/7b485022-353e-4fd4-b8a2-5befab213670">
<IMG title="AUGER;ZENITH" src="https://github.com/MrAlaux/Nugget-Doom/assets/74435632/b07e3fa9-01a9-4228-97bb-0d4bb89c589f">
<IMG title="Doom II" src="https://github.com/MrAlaux/Nugget-Doom/assets/74435632/21534ee9-7129-44c6-a62b-0c50f801a611">
</details>

_REQUIRES at least Nugget Doom 3.1.0..._

**Comment by Sarnion** [comment]
Here is the hud I play with. I have not tested it in wide screen.
[ammohud2024.zip](https://github.com/MrAlaux/Nugget-Doom/files/15412977/ammohud2024.zip)
![ammohud2024preview2](https://github.com/MrAlaux/Nugget-Doom/assets/144228028/deea05a8-5f63-4420-b520-6de147c10cd0)
![ammohud2024preview1](https://github.com/MrAlaux/Nugget-Doom/assets/144228028/d53b0ffa-9157-48f7-9d32-46538e15895c)

**Comment by sencuss** [comment]
changing this because theres more than 1 hud now
## **A Hud**
**_[nuggetdoom-ahud.zip](https://github.com/MrAlaux/Nugget-Doom/files/15451198/nuggetdoom-ahud.zip)_**

_hotfix to fix timer and powerups_
_hotfix 2 to fix automap title overlapping the timer (im good at testing)_

### features

- a slightly more traditional fps hud layout
- unity port assets for the K/I/S icons
- 4:3, 21:9, & 16:10 support

<details>
  <summary>preview</summary>
  
![nugg0017](https://github.com/MrAlaux/Nugget-Doom/assets/64239553/2e0d58b7-f80c-48ef-ba30-9a35c9e40437)
![nugg0018](https://github.com/MrAlaux/Nugget-Doom/assets/64239553/a9e7207c-676e-46c6-8a6b-3b760ce72b4f)
![nugg0019](https://github.com/MrAlaux/Nugget-Doom/assets/64239553/047058c4-f098-4d16-b1c4-a6d9193726b2)
</details>

## **PSX Hud**
**_[nuggetdoom-psxhud.zip](https://github.com/user-attachments/files/15523401/nuggetdoom-psxhud.zip)_**

### features
- a nearly perfect recreation of playstation doom's hud
- K/I/S placement based on psydoom
- 4:3 and 16:10 support

### issues
- currently doesn't support ultrawide without it cutting off (i'm kinda lazy to do that)
- no box around your current weapon (would be neat)
- message location isn't correct (x position doesn't work due to a bug with nugget)
- no proper way to override the arms box with frags for deathmatch, so it uses nugget's way of replacing K/I/S

### notes
- weapon numbers are intentionally out of order, honestly felt wrong with it being normal
- mugshot is NOT replaced, to let custom ones be used
- don't worry about slot 4 being missing in the screenshots, it works
<details>
  <summary>preview</summary>

![nugg0024](https://github.com/MrAlaux/Nugget-Doom/assets/64239553/f805bfd3-154a-4a7d-b42a-545cb36a33ec)
![nugg0026](https://github.com/MrAlaux/Nugget-Doom/assets/64239553/9979a33b-3a35-4366-bb89-02f36cb75b7e)
![nugg0027](https://github.com/MrAlaux/Nugget-Doom/assets/64239553/61decd58-e994-4577-bbfd-12bb7c584204)
</details>

**Comment by Duffking** [comment]
[nughud_dretro.zip](https://github.com/user-attachments/files/16666678/nughud_dretro.zip) (v2 - fixed rocket sprite alignment)

Tried to make something similar to Doom Retro's HUD which I really like for its minimalism while keeping the mugshot. 

Shamelessy stole @d-bind's qlike HUD as a starting point, hope you don't mind. Knocked this together in about 30 minutes having never edited a WAD lump before so if there's any issues with it I apologise.

Edit: Berserk icon misalignment should be fixed with next version of the port!

<details>
<summary>Preview</summary>

![nugget-doom_NKlncwOBeJ](https://github.com/user-attachments/assets/238403fb-9fe0-4f07-b59c-f70ffc14b40b)

![nugget-doom_Cj4Lj2GNSI](https://github.com/user-attachments/assets/2ccfc8e3-5a98-487c-a585-1f7559425282)
</details>

**Comment by OnymynO** [comment]
[nughud.myno.zip](https://github.com/user-attachments/files/17925646/nughud.myno.zip)

My philosophy on this one:

- I wanted the most important info (health and armor numbers) nearest to the center.
- Icons to tell the kind of armor you have and whether you have Berserk is a must.
- Ammo counter is disabled; it is redundant when you have the full ammo tally.
- Which keys you have should be grouped with arms (which weapons you have) and out of the way.
- The face is here; I missed it.
- Room for HUD messages and typing multiplayer text without anything getting in the way.

<details>
  <summary>Preview 1:</summary>
  
  ![nughud myno 1](https://github.com/user-attachments/assets/44e4307a-971d-4419-8069-a30b543487b2)
  
</details>

<details>
  <summary>Preview 2 (everything active in automap view):</summary>
  
  ![nughud myno 2](https://github.com/user-attachments/assets/072f786e-75fb-4336-9b92-1d0a50adadc0)

</details>

**Comment by PorpiangWriter** [comment]
[NUGHUD-Vertical Widget.zip](https://github.com/user-attachments/files/20381399/NUGHUD-Vertical.Widget.zip)
![Vertical Widget NUGHUD](https://github.com/user-attachments/assets/cc957d9f-2ccd-44f3-aa86-00a68c12291d)
Okay, here's my attempt to adjust NUGHUD's widget, although I can't do it vertically, but at least I placed it in the same place as the default SBARDEFs.

Update:
[NUGHUD-Total Vertical Widget.zip](https://github.com/user-attachments/files/20503441/NUGHUD-Total.Vertical.Widget.zip)
![Total Vertical Widget NUGHUD](https://github.com/user-attachments/assets/17e28e8a-0981-4ce9-ac95-d1aeb296befe)
Also, another attempt to make NUGHUD's widgets vertical by the power of nughud_[something]_ml, featuring mugshot!

Update 2:
![New Total Vertical Widget NUGHUD](https://github.com/user-attachments/assets/5eadc73a-218f-46ab-9f75-66aade82b3c4)
[NUGHUD-Total Vertical Widget.zip](https://github.com/user-attachments/files/20649582/NUGHUD-Total.Vertical.Widget.zip)
And here we are, more updated, as close to the SBARDEF widget as possible, with a fist weapon indicator.

**Comment by mtslzr** [comment]
[nughud.mtslzr.zip](https://github.com/user-attachments/files/30543724/nughud.mtslzr.zip)

Not hugely unique, but I started with [myno](https://github.com/MrAlaux/Nugget-Doom/discussions/54#discussioncomment-11388335) and then tweaked the positioning to be closer to [Neon Overdrive](https://www.doomworld.com/forum/topic/152555-neon-overdrive-mbf21umapinfo-idgames/).

I tried to go for balanced and evenly-spaced, including the automap widgets. Keeps health/armor/ammo easy to see at a glance.

<img width="3440" height="1440" alt="nughud" src="https://github.com/user-attachments/assets/2e7ca00d-69e0-47c5-94f2-3c0f5e34300f" />

<img width="3440" height="1440" alt="nughud-automap" src="https://github.com/user-attachments/assets/01edf88f-0bb5-48f6-89b6-651d4dfed7b0" />

**Comment by MrAlaux** [comment]
Posting on behalf of _lilywaves_:

### [lilyhud_nug](https://github.com/user-attachments/files/30616567/lilyhud_nug.zip)

Based on _wideHUD_ as seen above.

<details><summary>Preview</summary>
<p>
<img width="1366" height="768" alt="imagen" src="https://github.com/user-attachments/assets/069d386e-be76-43cc-8822-10a826da1446" />
</p>
</details>
